File: Iverson.Clients/DotNet/Iverson.Client.Conformance.Driver/Models/ErrorUnregisteredDoc.cs
using Iverson.Client.Attributes;

namespace Iverson.Client.Conformance.Driver.Models;

/// <summary>
/// S9 <c>error-contract</c>'s unregistered fixture: declared by all five drivers through their own
/// client libraries and registered by NOTHING — no driver, no scenario, no orchestrator, in this
/// run or any other. A mapped write against it must be refused with <c>FailedPrecondition</c>
/// (<c>ObjectMappingGrpcService.RequireSchema</c>), which is the whole observation.
///
/// <para><b>Do not register this type.</b> Every register phase in this driver sets
/// <c>capture.OnlySendTypeName</c> before calling <c>SchemaRegistrar.RegisterAllAsync()</c>, so the
/// registrar's walk over the whole assembly's <c>[IversonEntity]</c> types never actually sends
/// this one. Removing that guard from any register phase would register it as a side effect and
/// silently turn <c>IVC-ERR-005</c> green for the wrong reason — the write would then be accepted,
/// which the assertion reads as a failure, so the harness would go red rather than lie, but the
/// fixture would be gone.</para>
/// </summary>
[IversonEntity]
public class ErrorUnregisteredDoc
{
    [IversonKey] public Guid Id { get; set; }
    [IversonTenant] public string TenantId { get; set; } = string.Empty;
    public string OwnerId { get; set; } = string.Empty;
    public string Label { get; set; } = string.Empty;
}

File: Iverson.Clients/Java/conformance/src/main/java/io/iverson/conformance/models/ErrorUnregisteredDoc.java
package io.iverson.conformance.models;

import io.iverson.client.annotations.IversonEntity;
import io.iverson.client.annotations.IversonKey;

import java.util.UUID;

/**
 * S9 {@code error-contract}'s unregistered fixture: declared by all five drivers and registered by
 * NOTHING — no driver, no scenario, no orchestrator, in this run or any other. A mapped write
 * against it must be refused with {@code FAILED_PRECONDITION}
 * ({@code ObjectMappingGrpcService.RequireSchema}), which is the whole observation.
 *
 * <p>Do not hand this type to any {@code SchemaRegistrar} call. Registering it would destroy the
 * fixture {@code IVC-ERR-005} depends on.
 */
@IversonEntity
public class ErrorUnregisteredDoc {

    @IversonKey
    private UUID id;

    private String tenantId;

    private String ownerId;
    private String label;

    public UUID getId() { return id; }
    public void setId(UUID id) { this.id = id; }

    public String getTenantId() { return tenantId; }
    public void setTenantId(String tenantId) { this.tenantId = tenantId; }

    public String getOwnerId() { return ownerId; }
    public void setOwnerId(String ownerId) { this.ownerId = ownerId; }

    public String getLabel() { return label; }
    public void setLabel(String label) { this.label = label; }
}
